From: Andrea Righi <andrea.righi@canonical.com>
Subject: mei_cl_bus: fix build error with kernels > 5.13
Description: Apparently mei_cldev_dma_map/unmap() are not available on kernels
 greater than 5.13, as incorrectly assumed by the following condition:
 
  #if LINUX_VERSION_IS_LESS(5,13,0) && !defined(CONFIG_WLAN_VENDOR_INTEL_MEI)
 
 Simply rely on CONFIG_WLAN_VENDOR_INTEL_MEI to determine when
 mei_cldev_dma_map/unmap() should be defined in the kernel.
Author: Andrea Righi <andrea.righi@canonical.com>
Bug: https://gitlab.com/vicamo/backport-iwlwifi-dkms/-/issues/12

Signed-off-by: Andrea Righi <andrea.righi@canonical.com>
---
 backport-include/linux/mei_cl_bus.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/backport-include/linux/mei_cl_bus.h b/backport-include/linux/mei_cl_bus.h
index be5b262ad..0c6983122 100644
--- a/backport-include/linux/mei_cl_bus.h
+++ b/backport-include/linux/mei_cl_bus.h
@@ -22,7 +22,7 @@
 #define mei_cldev_disable(cldev) mei_cl_disable_device(cldev)
 #endif
 
-#if LINUX_VERSION_IS_LESS(5,13,0) && !defined(CONFIG_WLAN_VENDOR_INTEL_MEI)
+#ifndef CONFIG_WLAN_VENDOR_INTEL_MEI
 static inline void *mei_cldev_dma_map(struct mei_cl_device *cldev,
 				      u8 buffer_id, size_t size)
 {
-- 
2.33.1

